forest/daemon/src/daemon.rs

```diff
@@ -377,6 +380,15 @@ pub(super) async fn start(config: Config, detached: bool) -> Db {
 
     services.spawn(p2p_service.run());
 
+    while let Some(res) = services.join_next().await {
+        if let Ok(res_inner) = res {
+            if let Err(error_message) = res_inner {
+                let msg = format!("services failure: {}", error_message);
+                cli_error_and_die(msg, 1);
+            }
+        }
+    }
+
     // Block until ctrl-c is hit
     ctrlc_oneshot.await.unwrap();
```

Comment:
I don't think this works. You're no longer listening for ctrl-c.

Reply:
I think I fixed it

Reply:
Refixed using `select!`